File: be/src/main/java/com/project/popupmarket/dto/oauth/LoginResponse.java
package com.project.popupmarket.dto.oauth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@AllArgsConstructor
@Builder
public class LoginResponse {
    private final String accessToken;
    private final String refreshToken;
    private final boolean requireSignup;
    private final String signupUuid;

    public static LoginResponse requireSignup(String uuid) {
        return new LoginResponse(null, null, true, uuid);
    }

    public LoginResponse(String accessToken, String refreshToken) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.requireSignup = false;
        this.signupUuid = null;
    }
}

File: be/src/main/java/com/project/popupmarket/dto/oauth/LoginResponse.kt
package com.project.popupmarket.dto.oauth

data class LoginResponse(
    val accessToken: String? = null,
    val refreshToken: String? = null,
    val isRequireSignup: Boolean = false,
    val signupUuid: String? = null
) {
    companion object {
        @JvmStatic
        fun requireSignup(uuid: String): LoginResponse {
            return LoginResponse(
                isRequireSignup = true,
                signupUuid = uuid
            )
        }
    }

    constructor(accessToken: String, refreshToken: String) : this(
        accessToken = accessToken,
        refreshToken = refreshToken,
        isRequireSignup = false,
        signupUuid = null
    )
}